src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -62,9 +77,16 @@ public List<TElement> ToList()
             if (count > 0)
             {
                 int[] map = SortedMap(buffer);
-                for (int i = 0; i != count; i++)
+                try
                 {
-                    list.Add(buffer._items[map[i]]);
+                    for (int i = 0; i != count; i++)
+                    {
+                        list.Add(buffer._items[map[i]]);
+                    }
+                }
+                finally
```

Comment:
Likewise on the try…finally. `list.Add()` can throw if it allocates, but we've preallocated already.